h2o-extensions/target-encoder/src/test/java/ai/h2o/targetencoding/strategy/TargetEncoderRGSTest.java

```diff
@@ -52,4 +56,82 @@ public void getTargetEncodingMapByTrainingTEBuilder() {
     }
   }
 
+  @Test
+  public void regularGSOverTEParameters() {
+
+    Scope.enter();
+    try {
+      Frame trainingFrame = parse_test_file("./smalldata/gbm_test/titanic.csv");
+      Scope.track(trainingFrame);
+      String responseColumn = "survived";
+      asFactor(trainingFrame, responseColumn);
+
+      HashMap<String, Object[]> hpGrid = new HashMap<>();
+      hpGrid.put("blending", new Boolean[]{true, false});
+      hpGrid.put("noise_level", new Double[]{0.0, 0.01,  0.1});
+      hpGrid.put("k", new Double[]{1.0, 2.0, 3.0});
+      hpGrid.put("f", new Double[]{5.0, 10.0, 20.0});
+
+      TargetEncoderModel.TargetEncoderParameters parameters = new TargetEncoderModel.TargetEncoderParameters();
+      parameters._train = trainingFrame._key;
+      parameters._response_column = responseColumn;
+      parameters._ignored_columns = ignoredColumns(trainingFrame, "home.dest", "embarked", parameters._response_column);
+
+      GridSearchHandler.DefaultModelParametersBuilderFactory<TargetEncoderModel.TargetEncoderParameters, TargetEncoderV3.TargetEncoderParametersV3> modelParametersBuilderFactory = new GridSearchHandler.DefaultModelParametersBuilderFactory<>();
+
+      HyperSpaceSearchCriteria.RandomDiscreteValueSearchCriteria hyperSpaceSearchCriteria = new HyperSpaceSearchCriteria.RandomDiscreteValueSearchCriteria();
+      HyperSpaceWalker.RandomDiscreteValueWalker<TargetEncoderModel.TargetEncoderParameters> walker = new HyperSpaceWalker.RandomDiscreteValueWalker<>(parameters, hpGrid, modelParametersBuilderFactory, hyperSpaceSearchCriteria);
+
+      Job<Grid> gs = GridSearch.startGridSearch(null, walker, 1);
+
+      Scope.track_generic(gs);
+      final Grid grid = gs.get();
+      Scope.track_generic(grid);
+
+      assertEquals(54, grid.getModelCount());
+
+    } finally {
+      Scope.exit();
+    }
+  }
+
+  @Test // Probably could be resolved with redesign/fix PUBDEV-7110
```

Comment:
This parallel version of test is not fixed yet. Will try to merge in https://github.com/h2oai/h2o-3/pull/4191/files to see if it fixes issue.